ehrql/tables/beta/tpp.py

```diff
@@ -586,129 +671,225 @@ class occupation_on_covid_vaccine_record(EventFrame):
 
 @table
 class opa(EventFrame):
+    """
+    Outpatient appointments data (OPA) is provided via the NHS Secondary Uses Service.
+
+    This table gives core details of outpatient appointments.
+
+    Refer to the GitHub issue that [describes limitations
+    of the outpatient appointments data][opa_limitations_issue]
+    and the GitHub issue that [discusses more of the background context][opa_context_issue].
+
+    [opa_limitations_issue]: https://github.com/opensafely-core/cohort-extractor/issues/673
+    [opa_context_issue]: https://github.com/opensafely-core/cohort-extractor/issues/492
+    """
+
     opa_ident = Series(
         int,
         constraints=[Constraint.NotNull()],
-        description="TODO",
+        description="Unique identifier for the appointment used across the OPA tables.",
     )
     appointment_date = Series(
         datetime.date,
-        description="TODO",
+        description="The date of an appointment.",
     )
     attendance_status = Series(
         str,
-        description="TODO",
+        description=(
+            "Indicates whether or not an appointment for a care contact took place. "
+            "If the appointment did not take place it also indicates whether or not advanced warning was given. "
+            "Refer to the [NHS Data Model and Dictionary entry for attendance status]"
+            "(https://www.datadictionary.nhs.uk/data_elements/attendance_status.html) "
+            "for details on code meanings."
+        ),
+        # This non-ascending order follows the NHS Data Model and Dictionary.
+        constraints=[Constraint.Categorical(["5", "6", "7", "2", "3", "4"])],
     )
     consultation_medium_used = Series(
         str,
-        description="TODO",
+        description=(
+            "Identifies the communication mechanism used to relay information "
+            "between the care professional and the person who is the subject of the consultation, "
+            "during a care activity. "
+            "Refer to the [NHS Data Model and Dictionary entry for consultation medium used]"
+            "(https://www.datadictionary.nhs.uk/data_elements/consultation_medium_used.html) "
+            "for details on code meanings."
+        ),
+        constraints=[
+            Constraint.Categorical(
+                ["01", "02", "03", "04", "05", "06", "07", "08", "98"]
+            )
+        ],
```

Comment:
comment: The same warning about these being taken from the NHS reference, and not verified myself in the TPP data applies: https://github.com/opensafely-core/ehrql/pull/1628/files#r1386573765

Reply:
The values in the DB are `01`, `02`, `03`, `04`, `05`, `09`, `10`, `11`, `98`, and `NULL`.